src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -1813,40 +1813,40 @@ bool workflow_set_sandbox(ADUC_WorkflowHandle handle, const char* sandbox)
     return true;
 }
 
-bool _workflow_copy_config_download_folder(char* dest, const size_t dest_size)
+char* _workflow_copy_config_downloads_folder(const size_t max_size)
 {
-    bool success = false;
+    char* downloads_folder_path = NULL;
+
     const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
     if (config == NULL)
     {
         goto done; /* Config is unitialized or we could not get a reference */
     }
 
-    size_t download_folder_len = ADUC_StrNLen(config->downloadsFolder, dest_size);
+    size_t download_folder_len = ADUC_StrNLen(config->downloadsFolder, max_size);
 
-    if (download_folder_len == 0 || download_folder_len == dest_size)
+    if (download_folder_len == 0 || download_folder_len == max_size)
     {
         Log_Error("Invalid base sandbox dir: '%s'", config->downloadsFolder);
         goto done;
     }
 
-    if (ADUC_Safe_StrCopyN(dest, config->downloadsFolder, dest_size, download_folder_len) != download_folder_len)
+    if (!mallocAndStrcpy_s(&downloads_folder_path, config->downloadsFolder))
     {
-        Log_Error("Failed copy config download folder");
+        Log_Error("Failed to allocate memory for downloads folder path");
```

Comment:
Should probably omit this as the logging would likely fail in this case too unless logging optimistically?